containerd-shim-v2/wait.go

```diff
@@ -126,3 +127,26 @@ func watchSandbox(s *service) {
 	// Existing container/exec will be cleaned up by its waiters.
 	// No need to send async events here.
 }
+
+func watchOOMEvents(s *service) {
+	if s.sandbox == nil {
+		return
+	}
+
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		default:
+			containerID, err := s.sandbox.GetOOMEvent()
+			if err != nil {
+				logrus.WithError(err).Warn("failed to get oom event from sandbox")
```

Comment:
Is this for backward compatibility with an older agent?

Reply:
Yes it should - we agreed that the shim/runtime should just log an error/warning if it isn't able to get oom events from the agent, see https://github.com/kata-containers/runtime/issues/2150#issuecomment-559655563

The log line here might be a little verbose, maybe we can detect if the agent is older/doesn't support the OOM event request and exit this go routine early?

Reply:
Maybe we can leverage the error codes from grpc here:
https://github.com/grpc/grpc/blob/master/doc/statuscodes.md

I see 
```
UNIMPLEMENTED | 12 | The operation is not implemented or is not supported/enabled in this service.
```
If not, we can go ahead and add a comment here for backward compatibility.

Reply:
👍 Good suggestion. I've updated it to check the code from the GRPC error and if it code 12/Unimplemented, then will exit early out of the loop.